src/main.js

```diff
@@ -1,4 +1,10 @@
 import TelemetryApp from './app/TelemetryApp.svelte';
+import { authenticate } from './utils/auth';
+import { store } from './app/state/store';
```

Comment:
wondering if it might make sense to (1) have a `createStore` function a la GUD here, and then (2) have the store instantiation happen in the `authentication` success callback. I suspect we'll have patterns like this in other data products down the line so it'd be good to get this right now rather than later. As it stands now, GLAM's state management is pretty close to GUD's as it is, but the idea of making a `createStore` function came out of our refactor, and hasn't hit this project yet.

Reply:
GUD doesn't have a `createStore` method. Do you mean `getDefaultState`?

Reply:
Ah. GLAM does have a `createStore` function already. I think I see what you mean. I'll move it there.

Reply:
I think I've done what you suggested, but let's check back in on this.

Reply:
I'm going to revert my work on this. I tried the following:

### store.js

```javascript
[...]
export let store = {};
export function initStore() {
  store = createStore(initialState);
}
[...]
```

### main.js

```javascript
import TelemetryApp from './app/TelemetryApp.svelte';
import { authenticate } from './utils/auth';
import { store, initStore } from './app/state/store';

// eslint-disable-next-line import/no-mutable-exports
let telemetryApp;

authenticate((token) => {
  initStore();
  store.setField('token', token);

  telemetryApp = new TelemetryApp({
    target: document.body,
  });
});

export default telemetryApp;
```

But I was getting an error about `store.subscribe` not being a function. Is this along the lines of what you had in mind, at least?

Reply:
yep, something like this is what I had in mind, but I can see why this might not work. It might end up just making more sense to not do the store instantiation in the `authenticate` success callback but just do the `setField` there.

Reply:
Can you say more about why that approach may not work? Is there something inherently wrong with it?

I'll leave this file as-is for now but I'm happy to revisit if we find a way to do this better.

Reply:
Yep, let's just leave this as-is for now. I think your approach works great here.